rest_framework/routers.py

```diff
@@ -52,12 +52,27 @@ def __init__(self):
     def register(self, prefix, viewset, basename=None):
         if basename is None:
             basename = self.get_default_basename(viewset)
-        self.registry.append((prefix, viewset, basename))
+
+        if not self.basename_already_registered(basename):
+            self.registry.append((prefix, viewset, basename))
 
         # invalidate the urls cache
         if hasattr(self, '_urls'):
             del self._urls
 
+    def basename_already_registered(self, new_basename):
```

Comment:
have to think if the naming is alright, also if we should document about the new check methods

Reply:
Maybe a simple `is_already_registered` could suit better?

Reply:
Done